uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Service/CiviHRStatsGathererTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use CRM_HRCore_Service_CiviHRStatsGatherer as CiviHRStatsGatherer;
+use CRM_HRCore_CMSData_Variable_VariableServiceInterface as VariableServiceInterface;
+use Civi\Test\TransactionalInterface;
+use Civi\Test\HeadlessInterface;
+
+/**
+ * @group headless
+ */
+class CiviHRStatsGathererTest extends PHPUnit_Framework_TestCase
+  implements TransactionalInterface, HeadlessInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  public function testGatheringStats() {
```

Comment:
I got a bit lazy towards the end here. Updating test to check all data in the stats class so far

Reply:
Yes, we need fabricators for UF_Match, Task, Document and Assignment. For the T&A entities, let's follow the same pattern we've been using for the CiviHR extensions and add the fabricator to the extension owning the entities they create.

Now, for the records in the drupal database, I'm really not sure if we should have fabricators or not. Maybe we can simply mock the functions so that they return the data we need for the tests?

Reply:
Mocking drupal functions is not too bad for simpler functions, but in this case it's a `db_select` which returns a query object. I think it could get pretty complicated to mock it, plus this code is behind a `isDrupal` call which will return false in the unit tests.